src/overlays/actors/ovl_Bg_Fire_Wall/z_bg_fire_wall.c

```diff
@@ -32,41 +38,184 @@ const ActorInit Bg_Fire_Wall_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Fire_Wall/func_809ACB28.s")
+static CollisionCheckInfoInit sColChkInfoInit = { 1, 80, 100, MASS_IMMOVABLE };
+
+static TexturePtr sFlameTextures[] = { gFwallFireball0Tex, gFwallFireball1Tex, gFwallFireball2Tex, gFwallFireball3Tex,
+                                       gFwallFireball4Tex, gFwallFireball5Tex, gFwallFireball6Tex, gFwallFireball7Tex };
+
+void BgFireWall_Init(Actor* thisx, PlayState* play) {
+    BgFireWall* this = (BgFireWall*)thisx;
+
+    this->unk_14C = this->actor.params;
+    this->actor.scale.y = 0.005f;
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+    this->actor.scale.z = 0.12f;
+    this->actor.scale.x = 0.12f;
+    this->unk_15C = 0.09f;
+    this->unk_158 = 0.1f;
+    this->unk_160 = 300.0f;
+    this->texIndex = Rand_S16Offset(0, 7);
+    this->actor.flags |= ACTOR_FLAG_10;
+    this->collider.dim.pos.y = this->actor.world.pos.y;
+    this->actionFunc = func_809AC638;
+}
+
+void BgFireWall_Destroy(Actor* thisx, PlayState* play) {
+    BgFireWall* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+s32 func_809AC5C0(BgFireWall* thisx, PlayState* play) {
+    BgFireWall* this = THIS;
+    Actor* player = play->actorCtx.actorLists[2].first;
+    Vec3f sp1C;
+
+    Actor_OffsetOfPointInActorCoords(&this->actor, &sp1C, &player->world.pos);
+    if ((fabsf(sp1C.x) < this->unk_160) && (fabsf(sp1C.z) < (this->unk_160 + 20.0f))) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void func_809AC638(BgFireWall* this, PlayState* play) {
+    if ((this->unk_14C != 0) || (func_809AC5C0(this, play) != 0)) {
+        this->actor.draw = BgFireWall_Draw;
+        this->timer = 5;
+        this->actionFunc = func_809AC68C;
+    }
+}
+
+void func_809AC68C(BgFireWall* this, PlayState* play) {
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    if (this->timer == 0) {
+        this->actionFunc = func_809AC6C0;
+    }
+}
+
+void func_809AC6C0(BgFireWall* this, PlayState* play) {
+    if ((this->unk_14C != 0) || (func_809AC5C0(this, play))) {
+        this->unk_15C = 0.09f;
+        Math_StepToF(&this->actor.scale.y, this->unk_15C + this->unk_154, this->unk_158);
+    } else if (Math_StepToF(&this->actor.scale.y, 0.005f, this->unk_158)) {
+        this->actionFunc = func_809AC638;
+    } else {
+        this->timer = 0;
+    }
+}
+
+void func_809AC760(BgFireWall* this, PlayState* play) {
+    s16 phi_a3;
+
+    if (Actor_IsFacingPlayer(&this->actor, 0x4000) != 0) {
+        phi_a3 = this->actor.shape.rot.y;
+    } else {
+        phi_a3 = (this->actor.shape.rot.y + 0x8000);
+    }
+    func_800B8D98(play, &this->actor, gGameInfo->data[0x990] + 10.0f, phi_a3, gGameInfo->data[0x991] + 5.0f);
```

Comment:
You should use `./tools/regconvert.py` for those values
```suggestion
    func_800B8D98(play, &this->actor, BREG(48) + 10.0f, phi_a3, BREG(49) + 5.0f);
```

Reply:
sonic@Ale-PC:~/mm/tools$ python regconvert.py /home/sonic/mm/src/overlays/actors/ovl_Bg_Fire_Wall/z_bg_fire_wall.c
Traceback (most recent call last):
  File "/home/sonic/mm/tools/regconvert.py", line 106, in <module>
    main()
  File "/home/sonic/mm/tools/regconvert.py", line 83, in main
    index = parse_number(args.index, args.hex)        
  File "/home/sonic/mm/tools/regconvert.py", line 19, in parse_number
    return int(string, 0)
ValueError: invalid literal for int() with base 0: '/home/sonic/mm/src/overlays/actors/ovl_Bg_Fire_Wall/z_bg_fire_wall.c'

Reply:
You need to pass `--file` if you want to pass a file instead of an index. The script has a help screen, pass `-h` to read it